### ISSUE TITLE
Workflows - History Dataset Association - PostgreSQL error

### ISSUE THREAD

**Issue by selten** [galaxyproject/galaxy#8096]
```
Traceback (most recent call last):
  File "lib/galaxy/web/framework/decorators.py", line 283, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "lib/galaxy/webapps/galaxy/api/workflows.py", line 733, in invoke
    request_params=work_request_params
  File "lib/galaxy/workflow/run.py", line 111, in queue_invoke
    modules.populate_module_and_state(trans, workflow, workflow_run_config.param_map, allow_tool_state_corrections=workflow_run_config.allow_tool_state_corrections)
  File "lib/galaxy/workflow/modules.py", line 1484, in populate_module_and_state
    step_errors = module_injector.inject(step, step_args=step_args)
  File "lib/galaxy/workflow/modules.py", line 1467, in inject
    state, step_errors = module.compute_runtime_state(self.trans, step, step_args)
  File "lib/galaxy/workflow/modules.py", line 1108, in compute_runtime_state
    state, step_errors = super(ToolModule, self).compute_runtime_state(trans, step, step_updates)
  File "lib/galaxy/workflow/modules.py", line 244, in compute_runtime_state
    visit_input_values(self.get_runtime_inputs(), state.inputs, update_value, no_replacement_value=NO_REPLACEMENT)
  File "lib/galaxy/tools/parameters/__init__.py", line 167, in visit_input_values
    callback_helper(input, input_values, name_prefix, label_prefix, parent_prefix=parent_prefix, context=context)
  File "lib/galaxy/tools/parameters/__init__.py", line 130, in callback_helper
    new_value = callback(**args)
  File "lib/galaxy/workflow/modules.py", line 238, in update_value
    value, error = check_param(trans, input, step_updates.get(prefixed_name), context)
  File "lib/galaxy/tools/parameters/__init__.py", line 184, in check_param
    value = param.from_json(value, trans, param_values)
  File "lib/galaxy/tools/parameters/basic.py", line 1778, in from_json
    rval = trans.sa_session.query(trans.app.model.HistoryDatasetAssociation).get(value)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 924, in get
    ident, loading.load_on_pk_identity)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 1007, in _get_impl
    return db_load_fn(self, primary_key_identity)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/loading.py", line 250, in load_on_pk_identity
    return q.one()
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2954, in one
    ret = self.one_or_none()
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2924, in one_or_none
    ret = list(self)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 2995, in __iter__
    return self._execute_and_instances(context)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/orm/query.py", line 3018, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 948, in execute
    return meth(self, multiparams, params)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
    compiled_sql, distilled_params
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1200, in _execute_context
    context)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
    exc_info
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 265, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1193, in _execute_context
    context)
  File "GLX/.venv/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 509, in do_execute
    cursor.execute(statement, parameters)
DataError: (psycopg2.DataError) invalid input syntax for integer: "__class__"
LINE 3: WHERE history_dataset_association.id = '__class__'
                                               ^
 [SQL: 'SELECT history_dataset_association.id AS history_dataset_association_id, history_dataset_association.history_id AS history_dataset_association_history_id, history_dataset_association.dataset_id AS history_dataset_association_dataset_id, history_dataset_association.create_time AS history_dataset_association_create_time, history_dataset_association.update_time AS history_dataset_association_update_time, history_dataset_association.state AS history_dataset_association_state, history_dataset_association.copied_from_history_dataset_association_id AS history_dataset_association_copied_from_history_dataset_a_1, history_dataset_association.copied_from_library_dataset_dataset_association_id AS history_dataset_association_copied_from_library_dataset_d_2, history_dataset_association.name AS history_dataset_association_name, history_dataset_association.info AS history_dataset_association_info, history_dataset_association.blurb AS history_dataset_association_blurb, history_dataset_association.peek AS history_dataset_association_peek, history_dataset_association.tool_version AS history_dataset_association_tool_version, history_dataset_association.extension AS history_dataset_association_extension, history_dataset_association.parent_id AS history_dataset_association_parent_id, history_dataset_association.designation AS history_dataset_association_designation, history_dataset_association.deleted AS history_dataset_association_deleted, history_dataset_association.visible AS history_dataset_association_visible, history_dataset_association.extended_metadata_id AS history_dataset_association_extended_metadata_id, history_dataset_association.version AS history_dataset_association_version, history_dataset_association.hid AS history_dataset_association_hid, history_dataset_association.purged AS history_dataset_association_purged, history_dataset_association.hidden_beneath_collection_instance_id AS history_dataset_association_hidden_beneath_collection_ins_3, dataset_1.id AS dataset_1_id, dataset_1.create_time AS dataset_1_create_time, dataset_1.update_time AS dataset_1_update_time, dataset_1.state AS dataset_1_state, dataset_1.deleted AS dataset_1_deleted, dataset_1.purged AS dataset_1_purged, dataset_1.purgable AS dataset_1_purgable, dataset_1.object_store_id AS dataset_1_object_store_id, dataset_1.external_filename AS dataset_1_external_filename, dataset_1._extra_files_path AS dataset_1__extra_files_path, dataset_1.file_size AS dataset_1_file_size, dataset_1.total_size AS dataset_1_total_size, dataset_1.uuid AS dataset_1_uuid \nFROM history_dataset_association LEFT OUTER JOIN dataset AS dataset_1 ON dataset_1.id = history_dataset_association.dataset_id \nWHERE history_dataset_association.id = %(param_1)s'] [parameters: {'param_1': u'__class__'}] (Background on this error at: http://sqlalche.me/e/9h9h)
```

One of my users is trying to submit a workflow, the user is told it doesn't work and the stacktrace is shown above. It seems similar to https://github.com/galaxyproject/galaxy/issues/4106.

This happens in Galaxy 19.01.
Please let me know if more information is required.

**Comment by mvdbeek** [comment]
Fixed in https://github.com/galaxyproject/galaxy/pull/7543, if you pull you'll get the fix.

**Comment by mvdbeek** [comment]
If you're already on the tip of the branch and you still see this we'll need the workflow to debug this further.

**Comment by selten** [comment]
Indeed that fixed it, I wasn't at the tip of 19.01.
